src/wp-includes/functions.php

```diff
@@ -8509,3 +8509,69 @@ function is_php_version_compatible( $required ) {
 function wp_fuzzy_number_match( $expected, $actual, $precision = 1 ) {
 	return abs( (float) $expected - (float) $actual ) <= $precision;
 }
+
+/**
+ * Get supported types for meta and option values.
+ *
+ * This is a subset of the PHP data types that can be stored in the database.
+ *
+ * @since 6.3.0
+ *
+ * @return array List of supported value types.
+ */
+function wp_get_db_value_types() {
+	return array(
+		'boolean',
+		'integer',
+		'float',
+		'string',
+		'array',
+		'object',
+		'null', // TODO: needed?
+	);
+}
+
+/**
+ * Decode a value from the database.
+ *
+ * Convert the value from the database back to its original type.
+ *
+ * @since 6.3.0
+ *
+ * @param mixed  $value      Raw value from the database.
+ * @param string $value_type Optional. The expected value type.
+ * @return mixed The decoded value.
+ */
+function wp_decode_value_from_db( $value, $value_type = '' ) {
+	$origianl_value = $value;
```

Comment:
Small typo here `$origianl_value`, also present in the `apply_filters`

Reply:
Fixed, thanks for spotting it :)